src/responder/ifp/ifp_users.c

```diff
@@ -1128,7 +1128,7 @@ ifp_users_find_by_valid_cert_send(TALLOC_CTX *mem_ctx,
         goto done;
     }
 
-    state->extra_args = talloc_zero_array(state, const char *, 8);
+    state->extra_args = talloc_zero_array(state, const char *, 10);
```

Comment:
Do you think it would be possible to factor out a common helper, that prepares p11_child args, out of ifp/pamsrv_p11/ssh to avoid code copy&paste?

Reply:
I will take a look

Reply:
It looks that most code is preparing options and parsing the response if any but they quite differ from each other. I do not think that if will hake the code shorter or cleaner.

Reply:
I meant "preparing options" part.